client/lib/path-to-section/index.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Internal dependencies
+ */
+import config from 'config';
+import { find, startsWith } from 'lodash';
+
+/**
+ * Conditional dependency
+ */
+const sections = config( 'project' ) === 'wordpress-com'
+	? require( 'wordpress-com' )
+	: null;
+
+/*
+ * wordpress-com-aware implementation where
+ *
+ *  f( '/' ) === 'reader"
+ *  f( '/design' ) === 'themes'
+ *  f( '/me' ) === 'me"
+ *  f( '/me/account' ) === 'me"
+ *  f( '/read' ) === 'reader'
+ */
+export const wpcomImplementation = path => {
+	const match = find( sections, section =>
+			section.paths.some( sectionPath =>
+				startsWith( path, sectionPath ) ) );
+
+	return match && match.name;
+};
+
+/*
+ * Dead-simple alternative implementation where
+ *
+ * 	f( '/foo/bar' ) === 'foo'
```

Comment:
It might be useful to include what `'/'`will resolve to, as it is mentioned in wordpress-com implementation above and can be considered a special case with few logical variants like `false`, `null`, `''`. Here we have the last one, right?